### PR TITLE
Updated CONTRIBUTING guide to explain pull request protocol

### PR DESCRIPTION
**Pull request by coldav**
# Overview

Explain to users how to create a pull request and request CI

# Reason for change

We wish to be available for pull requests and are tightening up the requirements for things like CI

# Description of change

Updated section in CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -253,6 +253,15 @@ steps.
 * Wait for a project maintainer to make an approving review.
 * Wait for your pull request to be merged.
 
+For CI you will need to ping the oneAPI Construction Kit reviewers by adding
+@ock-reviewers as a comment (or @ock-workflow-reviewers if it is workflow
+related). The CI can then be approved based on the PR. Note that currently some
```
Comment (hvdijk) [inline CONTRIBUTING.md:258]: This shouldn't be necessary, I thought? https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#about-code-owners says "Code owners are automatically requested for review when someone opens a pull request that modifies code that they own." If that does not work yet, I think we are doing something wrong.

Comment (coldav) [reply CONTRIBUTING.md:258]: ah fair, maybe just keep the ping for merging?

Comment (coldav) [reply CONTRIBUTING.md]: fixed

Comment (hvdijk) [reply CONTRIBUTING.md]: Looks right, though it's something that's effectively an extra bullet point in the list just above. Not sure if there's a nice unobtrusive way of putting it in there.

Comment (coldav) [reply CONTRIBUTING.md]: it feels too big a paragraph to add in there - it does mention fixing test failures in the bullet point, so I think I'd rather leave it as is.

**Review by hvdijk — APPROVED** [review]

**Pull request merged**